### PR TITLE
render system message for retention policy changes

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

**Comment by joshblum** [comment]
@cecileboucheron @adamjspooner  let me know if you have any opinions/changes you'd like to see, here are a bunch of examples of this message in use:

1-1 chat with a few different policies:
screen shot 2019-02-08 at 2 38 29 pm

team channel with a few different policies:
screen shot 2019-02-08 at 2 38 33 pm
screen shot 2019-02-08 at 2 38 37 pm

team channel from a team reader view (cannot change the setting):
screen shot 2019-02-08 at 2 43 28 pm

team wide setting changed (only a message to `#general`)
screen shot 2019-02-08 at 2 49 34 pm

File: shared/chat/conversation/messages/system-change-retention/index.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      if (props.isTeam) {
+        return 'team'
+      } else {
+        return 'channel'
+      }
+  }
+  return 'conversation'
+}
+
+const InheritDescription = (props: {isInherit: boolean}) => {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/index.js]: i don't think all these components are really necessary, just put this into the parent or  you can just figure this out in the render and render it inline like
```
const inherit = props.isInherit ? ' ' : ''
...
{inherit}
...
```

File: shared/chat/conversation/messages/system-change-retention/index.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import UserNotice from '../user-notice'
+import {formatTimeForMessages} from '../../../../util/timestamp'
+import * as RPCChatTypes from '../../../../constants/types/rpc-chat-gen'
+
+type Props = {|
+  canManage: boolean,
+  isTeam: boolean,
+  isInherit: boolean,
+  membersType: RPCChatTypes.ConversationMembersType,
+  onClickUserAvatar: (username: string) => void,
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/index.js]: why bother having a username if its just the props.username. just make it () and pass the username in mergeProps

File: shared/chat/conversation/messages/system-change-retention/index.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      }
+      break
+  }
+  return ''
+}
+
+const ChangeRetention = (props: Props) => {
+  return (
+    <UserNotice
+      style={{marginTop: Styles.globalMargins.small}}
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/index.js]: styles should use Styles.styleSheetCreate (grep it)

File: shared/chat/conversation/messages/system-change-retention/index.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      onClickAvatar={() => props.onClickUserAvatar(props.user)}
+    >
+      <Kb.Text
+        type="BodySmallSemibold"
+        backgroundMode="Announcements"
+        style={{color: Styles.globalColors.black_50}}
+      >
+        {formatTimeForMessages(props.timestamp)}
+      </Kb.Text>
+      <Kb.Box style={{...Styles.globalStyles.flexBoxColumn, alignItems: 'center'}}>
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/index.js]: prefer box2

File: shared/chat/conversation/messages/system-change-retention/container.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+})
+
+const mergeProps = (stateProps, dispatchProps, ownProps) => ({
+  canManage: stateProps.canManage,
+  isInherit: stateProps.isInherit,
+  isTeam: stateProps.isTeam,
+  membersType: stateProps.membersType,
+  onClickUserAvatar: (username: string) => dispatchProps._onClickUserAvatar(username),
+  onManageRetention: () => dispatchProps._onManageRetention(ownProps),
+  policy: stateProps.policy,
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/container.js]: avoid passing immutable to components

**Comment by cecileboucheron** [comment]
This seems below redundant to me - could we only have the banner?
image

**Comment by joshblum** [comment]
@cecileboucheron great point, i'll remove

**Comment by joshblum** [comment]
@chrisnojima addressed feedback, ptal

File: shared/chat/conversation/messages/system-change-retention/container.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  }
+  return {
+    canManage: canManage,
+    isInherit: ownProps.message.isInherit,
+    isTeam: ownProps.message.isTeam,
+    membersType: ownProps.message.membersType,
+    policy: ownProps.message.policy,
+    timestamp: ownProps.message.timestamp,
+    user: ownProps.message.user,
+    you: state.config.username || '',
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/container.js]: don't need this || actually

File: shared/chat/conversation/messages/system-change-retention/container.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  }
+}
+
+const mapDispatchToProps = (dispatch, ownProps) => ({
+  _onClickUserAvatar: ownProps => {
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/container.js]: don't pass ownprops, pass user

File: shared/chat/conversation/messages/system-change-retention/container.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      ? dispatch(ProfileGen.createShowUserProfile({username: ownProps.message.user}))
+      : dispatch(
+          TrackerGen.createGetProfile({
+            forceDisplay: true,
+            ignoreCache: true,
+            username: ownProps.message.user,
+          })
+        )
+  },
+  _onManageRetention: ownProps =>
```
Comment (chrisnojima) [inline shared/chat/conversation/messages/system-change-retention/container.js]: same here, just convoid

File: shared/chat/conversation/messages/system-change-retention/container.js (hunk 6 above)
Comment (joshblum) [reply shared/chat/conversation/messages/system-change-retention/container.js]: done

File: shared/chat/conversation/messages/system-change-retention/container.js (hunk 7 above)
Comment (joshblum) [reply shared/chat/conversation/messages/system-change-retention/container.js]: done

File: shared/chat/conversation/messages/system-change-retention/container.js (hunk 8 above)
Comment (joshblum) [reply shared/chat/conversation/messages/system-change-retention/container.js]: done

**Pull request merged**